### PR TITLE
Support Python 3

### PR DESCRIPTION
**Pull request by mrterry**
Use six so ndkale works in both python 2 and python 3

- [x] Port main codebase to use six python2/python3 interop
- [x] Run Travis for 2.7 and 3.5
- [x] Replace gflags with argparse

### CODE REVIEW

**Comment by mrterry** [comment]
@wenbinf @kkaehler This is ready for review.

The "big" changes are that:
  1. `crypt.py` is more strict regarding text vs bytes and uses `future`'s bytes class (backported from py3)
  1. In python 3 you can't compare `None` and `int`
  1. gflags is dropped in favor of argparse
  1. The test suite now runs for by python 2.7 and 3.5.

**Comment by sabw8217** [comment]
all crypto stuff looks good to me :+1:

File: kale/crypt.py [hunk 1]
```diff
@@ -16,14 +16,17 @@
...
 import base64
+import codecs
 import logging
 import struct
+import builtins
 
+from Crypto.Cipher import AES
 from kale import settings
-
+import six
```
Comment (kkaehler) [inline kale/crypt.py]: import ordering is off

File: kale/crypt.py [hunk 2]
```diff
@@ -157,13 +215,13 @@ def urlsafe_decrypt(msg):
 def _pad(msg):
     """Pad the message with enough bytes to be a multiple of BLOCK_SIZE.
 
-    :param str msg: str or unicode message to be padded.
-    :return: the msg with padding added.
-    :rtype: str
+    Args:
+        msg: str or unicode message to be padded.
+    Returns:
```
Comment (kkaehler) [inline kale/crypt.py]: missing new line

File: kale/crypt.py [hunk 3]
```diff
@@ -176,15 +234,17 @@ def _get_padding_bytes(msg_length):
     so if we would apply no padding, we add padding of
     BLOCK_SIZE.
 
-    :param msg_length: the length of the message to pad.
-    :return: the bytes that should be appended to the message to pad it.
+    Args:
+        msg_length: the length of the message to pad.
+    Returns:
```
Comment (kkaehler) [inline kale/crypt.py]: ditto

**Comment by wenbinf** [comment]
:+1:  Looks good to me

**Comment by mrterry** [comment]
Just cleaning up history.

**Pull request merged**